src/UserService.Business/UserCreateCommand.cs

```diff
@@ -1,36 +1,46 @@
-﻿using System;
-using FluentValidation;
+﻿using FluentValidation;
 using LT.DigitalOffice.UserService.Business.Interfaces;
 using LT.DigitalOffice.UserService.Data.Interfaces;
 using LT.DigitalOffice.UserService.Mappers.Interfaces;
 using LT.DigitalOffice.UserService.Models.Db;
 using LT.DigitalOffice.UserService.Models.Dto;
 using Microsoft.AspNetCore.Mvc;
+using System;
 
 namespace LT.DigitalOffice.UserService.Business
 {
     public class UserCreateCommand : IUserCreateCommand
     {
-        private readonly IValidator<UserCreateRequest> validator;
         private readonly IUserRepository repository;
-        private readonly IMapper<UserCreateRequest, DbUser> mapper;
+        private readonly IValidator<UserRequest> validator;
+        private readonly IMapper<UserRequest, DbUser> mapperUser;
+        private readonly IMapper<UserRequest, DbUserCredentials> mapperUserCredentials;
 
         public UserCreateCommand(
-            [FromServices] IValidator<UserCreateRequest> validator,
             [FromServices] IUserRepository repository,
-            [FromServices] IMapper<UserCreateRequest, DbUser> mapper)
+            [FromServices] IValidator<UserRequest> validator,
+            [FromServices] IMapper<UserRequest, DbUser> mapperUser,
+            [FromServices] IMapper<UserRequest, DbUserCredentials> mapperUserCredentials)
         {
             this.validator = validator;
             this.repository = repository;
-            this.mapper = mapper;
+            this.mapperUser = mapperUser;
+            this.mapperUserCredentials = mapperUserCredentials;
         }
 
-        public Guid Execute(UserCreateRequest request)
+        public Guid Execute(UserRequest request)
         {
             validator.ValidateAndThrow(request);
-            var user = mapper.Map(request);
 
-            return repository.UserCreate(user);
+            var dbUser = mapperUser.Map(request);
+            var dBUserCredentials = mapperUserCredentials.Map(request);
+
+            dBUserCredentials.Id = Guid.NewGuid();
+            dBUserCredentials.UserId = dbUser.Id;
+
+            repository.CreateUserCredentials(dBUserCredentials);
```

Comment:
first create user and only on success -> create credentials

Reply:
Fixed